### PR TITLE
Update release instructions

### PR DESCRIPTION
**Pull request by psiinon**
I'm sure there will be more to add - please comment with suggestions / corrections...

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Comment by psiinon** [comment]
Can be reviewed and merged if ok now?

File: RELEASING.md [hunk 1]
```diff
@@ -19,11 +36,31 @@ development iteration.
...
+- [ ] Announce on
+  - [ ] ZAP User and Dev groups
+  - [ ] @zaproxy twitter account
+  - [ ] OWASP Slack
+- [ ] Release client APIs
+  - [ ] [Java](https://github.com/zaproxy/zap-api-java/blob/main/RELEASING.md)
+  - [ ] [Python](https://github.com/zaproxy/zap-api-python/blob/master/RELEASING.md)
+- [ ] Update major projects using ZAP
+  - [ ] Kali - [new issue](https://bugs.kali.org/)
+  - [ ] Flathub
```
Comment (kingthorin) [inline RELEASING.md:56]: Add link? https://github.com/flathub/org.zaproxy.ZAP

Comment (kingthorin) [reply RELEASING.md:56]: I think this one we just have to manually update things. I couldn't find any defined/automated process.

File: RELEASING.md [hunk 2]
```diff
@@ -19,11 +36,31 @@ development iteration.
...
+  - [ ] Snap
```
Comment (kingthorin) [inline RELEASING.md:57]: Add link? https://github.com/zaproxy/zaproxy/tree/main/snap
The README in that directory seems to cover what needs to be done.

File: RELEASING.md [hunk 3]
```diff
@@ -2,14 +2,31 @@
...
+
+### Release Process
+
+The following steps should be followed in the given order to release the main version:
+- [ ] Run the workflow [Prepare Release Main Version](https://github.com/zaproxy/zaproxy/actions/workflows/prepare-release-main-version.yml), to prepare the release. It creates a pull request updating the version and indicating the remaining release tasks;
+- [ ] Finish the tasks indicated in the pull request;
+- [ ] Merge the pull request, to create the tag and the draft release (done by [Release Main Version](https://github.com/zaproxy/zaproxy/actions/workflows/release-main-version.yml));
+- [ ] Verify the draft release;
+- [ ] Publish the release.
+- [ ] Update [README.md](https://github.com/zaproxy/zaproxy/blob/main/README.md) to point to the latest JavaDocs
```
Comment (thc202) [inline RELEASING.md:29]: Can be before the `Finish the tasks, …`, to be included in the tag.

File: RELEASING.md [hunk 4]
```diff
@@ -2,14 +2,31 @@
...
+  - Development / bug fix issue links can be generated using the [zap-admin](https://github.com/zaproxy/zap-admin) `generateReleaseNotes` task.
+  - Library changes can be determined by diffing [LEGALNOTICE.md](https://github.com/zaproxy/zaproxy/blob/main/LEGALNOTICE.md) with the version at the previous release.
+- [ ] Create the [zap-admin](https://github.com/zaproxy/zap-admin) version and news files
+- [ ] Prepare blog post
+
+### Release Process
+
+The following steps should be followed in the given order to release the main version:
+- [ ] Run the workflow [Prepare Release Main Version](https://github.com/zaproxy/zaproxy/actions/workflows/prepare-release-main-version.yml), to prepare the release. It creates a pull request updating the version and indicating the remaining release tasks;
+- [ ] Finish the tasks indicated in the pull request;
```
Comment (thc202) [inline RELEASING.md:25]: Add before, e.g.:
`- [ ] Update [SECURITY.md](https://github.com/zaproxy/zaproxy/blob/main/SECURITY.md) to mention the latest version.`

File: RELEASING.md [hunk 5]
```diff
@@ -2,14 +2,31 @@
...
- 4. Verify the draft release;
- 5. Publish the release.
+### Pre release
+
+These tasks use checkboxes so that they can be copied into an issue.
+
+- [ ] Update dependencies - these can be checked using the [zaproxy](https://github.com/zaproxy/zaproxy) `dependencyUpdates` gradle task
+- [ ] Update the [gettingStarted](https://github.com/zaproxy/zap-extensions/tree/main/addOns/gettingStarted) ODT document and regenerate the PDF.
+- [ ] Update the [MacOS JRE](https://github.com/zaproxy/zaproxy/blob/main/buildSrc/src/main/kotlin/org/zaproxy/zap/distributions.gradle.kts).
+- [ ] Update the [MacOS Coyyright statement](https://github.com/zaproxy/zaproxy/blob/main/zap/src/main/macOS/OWASP%20ZAP.app/Contents/Info.plist).
```
Comment (thc202) [inline RELEASING.md:12]: There's a typo in copyright.

File: RELEASING.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (thc202) [reply RELEASING.md:25]: Pull request open to remove them #7537.

Comment (thc202) [reply RELEASING.md:25]: Tweaked the existing issue, to list these tasks.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Pull request merged**